bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondBridgeHandler.java

```diff
@@ -0,0 +1,314 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.bondhome.internal.api.BPUPListener;
+import org.openhab.binding.bondhome.internal.api.BPUPUpdate;
+import org.openhab.binding.bondhome.internal.api.BondDeviceState;
+import org.openhab.binding.bondhome.internal.api.BondHttpApi;
+import org.openhab.binding.bondhome.internal.api.BondSysVersion;
+import org.openhab.binding.bondhome.internal.config.BondBridgeConfiguration;
+import org.openhab.binding.bondhome.internal.discovery.BondDiscoveryService;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BondBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sara Geleskie Damiano - Initial contribution
+ */
+@NonNullByDefault
+public class BondBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BondBridgeHandler.class);
+
+    // Get a dedicated threadpool for the long-running listener thread.
+    // Intent is to not permanently tie up the common scheduler pool.
+    private final ScheduledExecutorService bondScheduler = ThreadPoolManager.getScheduledPool("bondBridgeHandler");
+    private final BPUPListener udpListener;
+    private final BondHttpApi api;
+
+    private @Nullable BondBridgeConfiguration config;
+
+    private @Nullable BondDiscoveryService discoveryService;
+
+    private final Set<BondDeviceHandler> handlers = Collections.synchronizedSet(new HashSet<>());
+
+    public BondBridgeHandler(Bridge bridge) {
+        super(bridge);
+        udpListener = new BPUPListener(this);
+        api = new BondHttpApi(this);
+        logger.debug("Created a BondBridgeHandler for thing '{}'", getThing().getUID());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Not needed, all commands are handled in the {@link BondDeviceHandler}
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(BondBridgeConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(this::initializeThing);
```

Comment:
it will actually recover when the BPUPListener initialized in the constructor receives a packet from a now-alive bridge, and calls setBridgeOnline with the new IP. also note that any hostnames are converted to IPs in initializeThing, so the only time you could ever have DNS problems is when the thing is first created. Though what happens when updateConfiguration is called if the thing is created via files?